docs/reference/transforms/matrix.rst

```diff
@@ -0,0 +1,210 @@
+.. _matrix transforms:
+
+Matrix Transforms
+=================
+
+The :mod:`taskgraph.transforms.matrix` transforms can be used to split a base
+task into many subtasks based on a defined matrix.
+
+These transforms are useful if you need to have many tasks that are very
+similar except for some small configuration differences.
+
+Usage
+-----
+
+Add the transform to the ``transforms`` key in your ``kind.yml`` file:
+
+.. code-block:: yaml
+
+   transforms:
+     - taskgraph.transforms.matrix
+     # ...
+
+Then create a ``matrix`` section in your task definition, e.g:
+
+.. code-block:: yaml
+
+   tasks:
+     test:
+       matrix:
+         os: ["win", "mac", "linux"]
+
+       # rest of task definition
+
+This will split the ``test`` task into three; ``test-win``, ``test-mac`` and
+``test-linux``.
+
+Matrix with Multiple Rows
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can add as many rows as you like to the matrix, and every combination of
+tasks will be generated. For example, the following matrix:
+
+.. code-block:: yaml
+
+   tasks:
+     test:
+       matrix:
+         os: ["win", "mac", "linux"]
+         python: ["py312", "py311"]
+
+       # rest of task definition
+
+Will generate these tasks:
+
+- ``test-win-py312``
+- ``test-win-py311``
+- ``test-mac-py312``
+- ``test-mac-py311``
+- ``test-linux-py312``
+- ``test-linux-py311``
+
+Note that the name of the tasks will be built based on the order of rows in the
```

Comment:
Is this guaranteed? It seems like this may differ based on what the yaml parser does and/or the `dict` (unless these are loaded into OrderedDicts).

Reply:
Yes, as of Python 3.7 the dict class preserves insertion order, so there's usually no need for OrderedDicts anymore. PyYaml should also keep the order because of this